From a22a6af7edeab9e7f265d3b4a43a348361a266c3 Mon Sep 17 00:00:00 2001
From: woahwhattheheck <293286387+woahwhattheheck@users.noreply.github.com>
Date: Thu, 1 Oct 2026 11:47:23 +0000
Subject: [PATCH 1/2] Cast: Keep status in CastDevice and skip icon without
 path

The discovery constructor dropped the status flags read from the TXT
record and built an icon URL ending in "null" when a receiver did not
announce an icon path.
---
 .../main/java/com/google/android/gms/cast/CastDevice.java    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/play-services-cast/src/main/java/com/google/android/gms/cast/CastDevice.java b/play-services-cast/src/main/java/com/google/android/gms/cast/CastDevice.java
index f185c8a5..4416bb94 100644
--- a/play-services-cast/src/main/java/com/google/android/gms/cast/CastDevice.java
+++ b/play-services-cast/src/main/java/com/google/android/gms/cast/CastDevice.java
@@ -48,9 +48,12 @@ public class CastDevice extends AutoSafeParcelable {
         this.deviceVersion = deviceVersion;
         this.friendlyName = friendlyName;
         this.icons = new ArrayList<WebImage>();
-        this.icons.add(new WebImage(Uri.parse(String.format("http://%s:8008%s", this.address, iconPath))));
+        if (iconPath != null) {
+            this.icons.add(new WebImage(Uri.parse(String.format("http://%s:8008%s", this.address, iconPath))));
+        }
         this.modelName = modelName;
         this.capabilities = capabilities;
+        this.status = status;
     }
 
     /**
-- 
2.43.0

